### PR TITLE
DOC: Improved docs for `jnp.fft.fft2` and `jnp.fft.ifft2`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/fft.py [hunk 1]
```diff
@@ -444,15 +444,162 @@ def _fft_core_2d(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
   return _fft_core(func_name, fft_type, a, s, axes, norm)
 
 
-@implements(np.fft.fft2)
 def fft2(a: ArrayLike, s: Shape | None = None, axes: Sequence[int] = (-2,-1),
          norm: str | None = None) -> Array:
+  r"""Compute a two-dimensional discrete Fourier transform along given axes.
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:449]: I don't think a raw string is necessary here – I don't see any escapted characters in the docstring.

File: jax/_src/numpy/fft.py [hunk 2]
```diff
@@ -444,15 +444,162 @@ def _fft_core_2d(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
...
+
+  JAX implementation of :func:`numpy.fft.fft2`.
+
+  Args:
+    a: input array.
+    s: sequence of integers. Specifies the shape of the result. ``s`` should have
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:455]: This is misleading I think – it's not the shape of the resulting array, but rather the size of the output in each specified axis.

Also, we should say "optional length-2 sequence of integers"

File: jax/_src/numpy/fft.py [hunk 3]
```diff
@@ -444,15 +444,162 @@ def _fft_core_2d(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
...
+      same length as ``axes``. If not specified, it will default to the shape of
+      ``a`` along the specified ``axes``.
+    axes: sequence of integers, default=(-2,-1). Specifies the axes along which
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:458]: Optional length-2 sequence of integers

File: jax/_src/numpy/fft.py [hunk 4]
```diff
@@ -444,15 +444,162 @@ def _fft_core_2d(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
...
+  See also:
+    - :func:`jax.numpy.fft.fft`: Computes a one-dimensional discrete Fourier
+      transform.
+    - :func:`jax.numpy.fft.ifft`: Computes a one-dimensional inverse discrete
+      Fourier transform.
+    - :func:`jax.numpy.fft.ifft2`: Computes a two-dimensional inverse discrete
+      Fourier transform.
+    - :func:`jax.numpy.fft.fftn`: Computes a multidimensional discrete Fourier
+      transform.
+    - :func:`jax.numpy.fft.ifftn`: Computes a multidimensional inverse discrete
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:476]: I would remove `ifftn` and `ifft` from this list. I think of them as two-steps removed from this function.

File: jax/_src/numpy/fft.py [hunk 5]
```diff
@@ -444,15 +444,162 @@ def _fft_core_2d(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
...
 
-@implements(np.fft.fft2)
 def fft2(a: ArrayLike, s: Shape | None = None, axes: Sequence[int] = (-2,-1),
          norm: str | None = None) -> Array:
+  r"""Compute a two-dimensional discrete Fourier transform along given axes.
+
+  JAX implementation of :func:`numpy.fft.fft2`.
+
+  Args:
+    a: input array.
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:454]: Mention that it should have at least two dimensions

File: jax/_src/numpy/fft.py [hunk 6]
```diff
@@ -444,15 +444,162 @@ def _fft_core_2d(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
...
+    Array(True, dtype=bool)
+  """
   return _fft_core_2d('fft2', xla_client.FftType.FFT, a, s=s, axes=axes,
                       norm=norm)
 
-@implements(np.fft.ifft2)
+
 def ifft2(a: ArrayLike, s: Shape | None = None, axes: Sequence[int] = (-2,-1),
           norm: str | None = None) -> Array:
+  r"""Compute a two-dimensional inverse discrete Fourier transform.
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:532]: Same broad comments as for `fft2` above

**Review by jakevdp — CHANGES_REQUESTED** [review]
Thanks! A couple comments below

File: jax/_src/numpy/fft.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Removed

File: jax/_src/numpy/fft.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Modified as suggested

File: jax/_src/numpy/fft.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Done

File: jax/_src/numpy/fft.py (hunk 5 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Done

File: jax/_src/numpy/fft.py (hunk 6 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Done

**Comment by rajasekharporeddy** [comment]
Made the suggested changes and squashed into single commit.

File: jax/_src/numpy/fft.py [hunk 7]
```diff
@@ -553,15 +553,158 @@ def _fft_core_2d(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
...
 def fft2(a: ArrayLike, s: Shape | None = None, axes: Sequence[int] = (-2,-1),
          norm: str | None = None) -> Array:
+  """Compute a two-dimensional discrete Fourier transform along given axes.
+
+  JAX implementation of :func:`numpy.fft.fft2`.
+
+  Args:
+    a: input array. Must have ``a.ndim >= 2``.
+    s: optional length-2 sequence of integers. Specifies the size of the output
+      along each specified axis. ``s`` should have same length as that of ``axes``.
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:565]: We've already said ``s`` needs to be length-2, so no need to mention the length of ``axes``

File: jax/_src/numpy/fft.py [hunk 8]
```diff
@@ -553,15 +553,158 @@ def _fft_core_2d(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
...
+      If not specified, it will default to the shape of ``a`` along the specified
+      ``axes``.
+    axes: optional length-2 sequence of integers, default=(-2,-1). Specifies the
+      axes along which the transform is computed. Length of ``axes`` must be
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:569]: We've already said ``axes`` must be length 2, no need to repeat that

File: jax/_src/numpy/fft.py [hunk 9]
```diff
@@ -553,15 +553,158 @@ def _fft_core_2d(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
...
+      equal to 2. If not specified, the transform is computed along the last two
+      axes.
+    norm: string. The normalization mode. "backward", "ortho" and "forward" are
+      supported.
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:573]: Default is "backward"

File: jax/_src/numpy/fft.py [hunk 10]
```diff
@@ -553,15 +553,158 @@ def _fft_core_2d(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
...
+  Args:
+    a: input array. Must have ``a.ndim >= 2``.
+    s: optional length-2 sequence of integers. Specifies the size of the output
+      along each specified axis. ``s`` should have same length as that of ``axes``.
+      If not specified, it will default to the shape of ``a`` along the specified
+      ``axes``.
+    axes: optional length-2 sequence of integers, default=(-2,-1). Specifies the
+      axes along which the transform is computed. Length of ``axes`` must be
+      equal to 2. If not specified, the transform is computed along the last two
+      axes.
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:571]: Either remove mention of last two axes here, or remove `default=(-2, -1)` above. We only need to say it once

File: jax/_src/numpy/fft.py [hunk 11]
```diff
@@ -553,15 +553,158 @@ def _fft_core_2d(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
...
+    >>> x_fft2 = jnp.fft.fft2(x)
+    >>> jnp.allclose(x, jnp.fft.ifft2(x_fft2))
+    Array(True, dtype=bool)
+  """
   return _fft_core_2d('fft2', xla_client.FftType.FFT, a, s=s, axes=axes,
                       norm=norm)
 
-@implements(np.fft.ifft2)
+
 def ifft2(a: ArrayLike, s: Shape | None = None, axes: Sequence[int] = (-2,-1),
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:638]: Similar comments apply here

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: jax/_src/numpy/fft.py (hunk 11 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py:635]: Done

File: jax/_src/numpy/fft.py (hunk 10 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Removed the mention of last two axes.

File: jax/_src/numpy/fft.py (hunk 9 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Done

File: jax/_src/numpy/fft.py (hunk 8 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Removed the statement about `axes` length.

**Review by jakevdp — APPROVED** [review]
Looks great, thanks!
